src/pal/inc/pal.h

```diff
@@ -5901,16 +5893,34 @@ typedef VOID (PALAPI *PHARDWARE_EXCEPTION_HANDLER)(PAL_SEHException* ex);
 PALIMPORT
 VOID
 PALAPI
-PAL_SetHardwareExceptionHandler(IN PHARDWARE_EXCEPTION_HANDLER handler);
+PAL_SetHardwareExceptionHandler(
+    IN PHARDWARE_EXCEPTION_HANDLER exceptionHandler);
 
-#endif // __cplusplus
+class PAL_HardwareExceptionHolder
```

Comment:
HardwareExceptionHolder means "smart pointer that keeps HardwareException object alive" (in the current naming conventions).

I would call it maybe HardwareExceptionHandlingStateHolder. (I know it is a long name, but it is not going to be used in too many places.)